src/shared/components/VideoPreviewBase/VideoPreviewBase.tsx

```diff
@@ -197,17 +261,39 @@ const VideoPreviewBase: React.FC<VideoPreviewBaseProps> = ({
                   </Anchor>
                 ))}
               {showMeta && (
-                <MetaContainer main={main}>
+                <MetaContainer noMarginTop={showChannel === false} main={main}>
                   {isLoading ? (
                     <SpacedPlaceholder height={main ? 16 : 12} width={main ? '40%' : '80%'} />
                   ) : createdAt ? (
                     <MetaText variant="subtitle2" main={main} scalingFactor={scalingFactor}>
-                      {formatVideoViewsAndDate(views ?? null, createdAt, { fullViews: main })}
+                      {/* TODO: instead of createdAt add the lastUpdate from useDraft hook */}
+                      {videoPublishState === 'draft'
+                        ? `Last updated ${formatDateAgo(createdAt)}`
+                        : formatVideoViewsAndDate(views ?? null, createdAt, { fullViews: main })}
                     </MetaText>
                   ) : null}
                 </MetaContainer>
               )}
             </TextContainer>
+            {publisherMode && !isLoading && (
+              <ContextMenuContainer>
+                <KebabMenuIconContainer onClick={(e) => openContextMenu(e, 200)}>
+                  <KebabMenuIcon />
+                </KebabMenuIconContainer>
+                {/* TODO: menu to be integrated with the video publishing functions */}
```

Comment:
Maybe we should just provide a callback for all of those in the props? Then the consumer can take care of this by itself

Reply:
Added them but I'm not too happy with it atm, I think I'll end up lifting this component later to VideoPreview or something like that